test/e2e/provisioner/provisioner_test.go

```diff
@@ -668,17 +658,16 @@ var _ = Describe("Gateway provisioner", func() {
 				} else {
 					// Root proxy in non-watched namespace should fail
 					By(fmt.Sprintf("Expect namespace %s not to be watched by contour", t.namespace))
-					hr, ok := f.CreateHTTPRouteAndWaitFor(route, e2e.HTTPRouteIgnoredByContour)
-					require.True(f.T(), ok, fmt.Sprintf("httproute's is %v", hr))
```

Comment:
I did use it here for debugging purpose in case it fails.

But yeah it's doable just to add a function to get the resource, since most of the code base doesn't use this.